src/inmanta/compiler/__init__.py

```diff
@@ -18,7 +18,7 @@
 import logging
 import sys
 from itertools import chain
-from typing import TYPE_CHECKING, Any, Dict, List, Optional, Sequence, Set, Tuple
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Sequence, Set, Tuple
```

Comment:
We should use `collections.abc.Callable` for new code.
```suggestion
from collections import abc
from typing import TYPE_CHECKING, Any, Dict, List, Optional, Sequence, Set, Tuple
```